integration-tests/src/test/resources/queries/sys_segment_queries.json

```diff
@@ -0,0 +1,38 @@
+[
+  {
+    "sqlQuery": {
+      "query": "SELECT datasource, count(*) FROM sys.segments GROUP BY 1"
+    },
+    "expectedResults": [
+      {
+        "datasource": "wikipedia_editstream",
+        "EXPR$1": 1
+      },
+      {
+        "datasource": "wikipedia",
+        "EXPR$1": 1
+      },
+      {
+        "datasource": "twitterstream",
+        "EXPR$1": 3
+      }
+    ]
+  },
+  {
+    "sqlQuery": {
+      "query": "SELECT server_type FROM sys.servers"
+    },
+    "expectedResults": [
+      {
+        "server_type": "historical"
+      }
+    ]
+  },
+  {
+    "sqlQuery": {
+      "query": "SELECT * FROM sys.tasks where status='FAILED'"
+    },
+    "expectedResults": [
+    ]
```

Comment:
empty result is not much helpful for validation. 
It wold be better if we can submit a task and then query for the status of it.

Reply:
sounds good, I added another test to query sys table for batch index tasks